lib/vector/Vlib/geos.c

```diff
@@ -362,15 +362,27 @@ GEOSCoordSequence *V1_read_line_geos(struct Map_info *Map, long offset,
     else
         z = NULL;
 
-    if (0 >= dig__fread_port_D(x, n_points, &(Map->dig_fp)))
+    if (0 >= dig__fread_port_D(x, n_points, &(Map->dig_fp))) {
+        G_free((void *)x);
+        G_free((void *)y);
+        G_free((void *)z);
```

Comment:
`pseq` should also be free'd, adding code. In this case I'd suggest to avoid some repeated code by using goto to clean up before return.

```suggestion
        GEOSCoordSeq_destroy(pseq);
        pseq = NULL;
        goto free_return;
```

The above should be added to all three cases.

And add the goto label `free_return:` before the line with `G_free((void *)x);` at the end of the function.

Reply:
done changes